app/src/main/java/com/calendar/YearViewFragment.java

```diff
@@ -0,0 +1,114 @@
+package com.calendar;
+
+import static com.calendar.CalendarUtils.daysInMonthInYearArray;
+import static com.calendar.CalendarUtils.yearFromDate;
+
+import android.os.Bundle;
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.LinearLayout;
+import android.widget.TextView;
+
+import androidx.fragment.app.Fragment;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.recyclerview.widget.GridLayoutManager;
+import androidx.recyclerview.widget.RecyclerView;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+
+public class YearViewFragment extends Fragment implements CalendarAdapter.OnItemListener {
+
+    private TextView monthYearText;
+    private final ArrayList<RecyclerView> monthsRecyclerView = new ArrayList<>();
+    private RecyclerView januaryRecyclerView;
+    private RecyclerView februaryRecyclerView;
+    private RecyclerView marchRecyclerView;
+    private RecyclerView aprilRecyclerView;
+    private RecyclerView mayRecyclerView;
+    private RecyclerView juneRecyclerView;
+    private RecyclerView julyRecyclerView;
+    private RecyclerView augustRecyclerView;
+    private RecyclerView septemberRecyclerView;
+    private RecyclerView octoberRecyclerView;
+    private RecyclerView novemberRecyclerView;
+    private RecyclerView decemberRecyclerView;
+
+    public YearViewFragment() {
+    }
+
+    @Nullable
+    @Override
+    public View onCreateView(@NonNull LayoutInflater inflater,
+                             @Nullable ViewGroup container,
+                             @Nullable Bundle savedInstanceState) {
+        View view = inflater.inflate(R.layout.fragment_year_view, container, false);
+
+        monthsRecyclerView.add(view.findViewById(R.id.calendarRecyclerView1));
```

Comment:
Wouldn't be better to use loop with getIdentifier to clean the code?

Reply:
I don't see any possibility to use loop here, i just moved code to the new method.